### PR TITLE
[VBLOCKS-2797] Rebase user defined messages

### PR DESCRIPTION
**Pull request by mhuynh5757**
## Submission Checklist

 - [x] ~Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code~
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

This PR makes a merge commit from main to the user defined messages feature branch.

## Breakdown

- Merge main into feature branch.
- Resolve some merge conflicts and use patterns from main instead of the feature branch.

## Validation

- Went through unit tests and wrote new ones using patterns from main.
- Removed old unit tests.
- Manual testing on iOS and Android.

## Additional Notes

N/A

### CODE REVIEW

File: src/CallInvite.tsx [hunk 1]
```diff
@@ -411,43 +628,115 @@ export namespace CallInvite {
...
     /**
-     * CallInviteMessage received event listener. This should be the function signature of
-     * any event listener bound to the {@link (CallInvite:namespace).Event.MessageReceived} event.
+     * Accepted event listener. This should be the function signature of any
+     * event listener bound to the {@link (CallInvite:namespace).Event.Accepted}
+     * event.
      *
      * @remarks
-     * See {@link (CallInvite:interface).(addListener:1)}.
+     * See {@link (Call:interface).(addListener:1)}.
```
Comment (kpchoy) [inline src/CallInvite.tsx:699]:
```suggestion
     * See {@link (CallInvite:interface).(addListener:1)}.
```

File: src/CallInvite.tsx [hunk 2]
```diff
@@ -411,43 +628,115 @@ export namespace CallInvite {
...
+     */
+    export type Rejected = () => void;
+
+    /**
+     * Cancelled event listener. This should be the function signature of any
+     * event listener bound to the
+     * {@link (CallInvite:namespace).Event.Cancelled} event.
+     *
+     * @remarks
+     * See {@link (Call:interface).(addListener:3)}.
```
Comment (kpchoy) [inline src/CallInvite.tsx:719]: Same as above, should this be `CallInvite:interface`

File: src/CallInvite.tsx [hunk 3]
```diff
@@ -411,43 +628,115 @@ export namespace CallInvite {
...
+     */
+    export type Cancelled = (error?: TwilioError) => void;
+
+    /**
+     * Rejected event listener. This should be the function signature of any
+     * event listener bound to the
+     * {@link (CallInvite:namespace).Event.NotificationTapped} event.
+     *
+     * @remarks
+     * See {@link (Call:interface).(addListener:4)}.
```
Comment (kpchoy) [inline src/CallInvite.tsx:729]: Same as above, should this be `CallInvite:interface`

**Review by kpchoy — CHANGES_REQUESTED** [review]

File: src/CallInvite.tsx [hunk 4]
```diff
@@ -55,46 +72,147 @@ export declare interface CallInvite {
...
+  /** {@inheritDoc (CallInvite:interface).(addListener:4)} */
+  on(
+    acceptedEvent: CallInvite.Event.NotificationTapped,
+    listener: CallInvite.Listener.NotificationTapped
+  ): this;
+
+  /**
+   * MessageReceived event. Raised when {@link (CallMessage:class)} is received.
+   * @example
+   * ```typescript
```
[the remark this answers is not in the document]
Comment (mhuynh5757) [reply src/CallInvite.tsx:197]: Good point, I'll add it for clarity's sake.

**Comment by mhuynh5757** [comment]
Good catch with the docstrings.

**Review by kpchoy — APPROVED** [review]
LGTM

File: src/CallInvite.tsx [hunk 5]
```diff
@@ -55,46 +72,149 @@ export declare interface CallInvite {
...
+    listener: CallInvite.Listener.Rejected
+  ): this;
+
+  /**
+   * Cancelled event. Raised when the call invite has been cancelled.
+   *
+   * @example
+   * ```ts
+   * voice.on(Voice.Event.CallInvite, (callInvite) => {
+   *   callInvite.on(CallInvite.Event.Cancelled, () => {
```
Comment (kpchoy) [inline src/CallInvite.tsx:140]: Do you want to add error here as argument?
```suggestion
   *   callInvite.on(CallInvite.Event.Cancelled, (error) => {
```

File: src/CallInvite.tsx [hunk 6]
```diff
@@ -55,46 +72,149 @@ export declare interface CallInvite {
...
+   * Accepted event. Raised when the call invite has been accepted.
+   *
    * @example
-   * ```typescript
-   * callInvite.addListener(CallInvite.Event.MessageReceived, (message) => {
-   *    // callMessage received
-   * })
+   * ```ts
+   * voice.on(Voice.Event.CallInvite, (callInvite) => {
+   *   callInvite.on(CallInvite.Event.Accepted, () => {
```
Comment (kpchoy) [inline src/CallInvite.tsx:80]: Do you want to add `call` here as argument? 
```suggestion
   *   callInvite.on(CallInvite.Event.Accepted, (call) => {
```

**Review by kpchoy — APPROVED** [review]

**Pull request merged**